### ISSUE TITLE
Use es6-standard module support

### ISSUE THREAD

**Issue by applmak** [google/chicago-brick#163]
Now that es6 modules are supported by default in Chrome 61 (and with the experimental web tech flag in M60), we should change our module APIs to use it by default in the browser. Nodejs support is still forthcoming, though it's expected to land behind a flag by the end of 2017.

**Comment by QuentinVecchio** [comment]
Hey, we're a group of 5 students from KTH looking to contribute to an open source project for a course and would like to try and convert the module to the new ES6 syntax if you are still interesting.

**Comment by applmak** [comment]
Hi! I'd be supportive of your doing this work.

To simplify things, you should focus on the code in `server/` or `client/` as that is far less complicated than code in, say, `lib/`, or `demo_modules/`.

I prefer small CLs that are easy to review. Do you think you'll be able to upgrade the code one file at a time?

Thanks so much for asking to help out!

**Comment by applmak** [comment]
This is now possible for everything in `server/`:

- Node now supports loading .mjs files via a hacky, non-standard way that they want to make more standard. See: https://github.com/nodejs/node-eps/blob/master/002-es-modules.md. Future versions of this support will definitely change, though, as nodejs moves toward implementing 'phase-0': https://github.com/nodejs/modules/blob/master/doc/plan-for-new-modules-implementation.md
- The docs are here: https://nodejs.org/api/esm.html. Because the current version doesn't support relative imports, we need to add a custom loader, which is a *very helpful thing* that browsers do not yet support. I prototyped this in our codebase, and it seems to work okay.
- It's possible to mix-and-match require and import statements, so an incremental conversion is possible.

`client/` and `lib/` and not yet able to be converted, because the browser support for modules isn't quite as powerful as node's, due to the lack of custom loader support. We might choose to hold out until this is all ready, or push forward with a server conversion now. I'd recommend holding off.

**Comment by applmak** [comment]
I've been thinking about this more, and I think it could be possible. Here's what has to happen:

1. The format of the slideshow modules has to change (again). The reason is that we lose the ability to inject any magical names into the global namespace (such as `register`). We need to create some other way by which we 'discover' the various symbols we need in a module. If we force es6-style js modules, we'll be able to list a set of exports and require that such exports include a specific name. For example:

```javascript
function load(...) {
  return {server: ..., client: ...};
}

exports = {load};
```

I like this because it allows the framework to inspect `load` and inject various named parameters and make them module-scope. I am confident in the browser's ability to make this work, given the current dynamic 'import' statement's semantics. It's less clear to me if node will be as flexible, but if necessary, node could read the file, dynamically add a umd wrapper around the `exports` line, and eval it. This should get us es6-style modules and hopefully simplify the framework a whole bunch.

**Comment by applmak** [comment]
Alright, I've created a prototype of this, and I believe that it's technically possible to pull off, and probably worth doing. Thoughts:
- This requires node 12+, because it's the only version of node that will load .js files as es6 files. You need to add an annoying package.json file that contains `{type: "module"}` to `demo_modules/` or to each module directory.
- It significantly simplifies a lot of the module loading.
- It *might* be able to be done in smaller steps (first, converting the demo_modules to es6; next, switching to `import()` to load them) via [`vm.SourceTextModule`](https://nodejs.org/api/vm.html#vm_class_vm_sourcetextmodule), but I don't think there's a whole lot of value to doing that.

**Comment by applmak** [comment]
This is complete @ head and will be released in 0.5.0.
